src/merge.py

```diff
@@ -15,8 +15,31 @@ def merge(x: list[int], y: list[int]) -> list[int]:
     z = []  # a new list to copy elements into
     # FIXME: fill out the loop so you merge the lists
     # until one of them is empty
+    # if (bool(x) and bool(y)) == False:
+    #    return x or y
+
+    # sooo, dont do both at the same time. is bad
     while i < len(x) and j < len(y):
-        break  # FIXME: you shouldn't just break here
-    # At least one of the lists is empty now. Copy the
+
+        if x[i] < y[j]:
+            z.append(x[i])
+            i += 1
+        else:
+            z.append(y[j])
+            j += 1
+
+        # FIXME: you shouldn't just break here
+        # At least one of the lists is empty now. Copy the
     # remainder of the other into z.
+    if i < len(x):
+        z.extend(x[i:])
+    if j < len(y):
+        z.extend(y[j:])
```

Comment:
You don't actually have to check `k < len(z)` here. If you are at the length, the slice `z[k:]` is empty, and it is perfectly valid to `extend` with an empty list.